shell/initialize/client.js

```diff
@@ -432,52 +265,6 @@ function checkForErrors(app) {
   }
 }
 
-// When navigating on a different route but the same component is used, Vue.js
-// Will not update the instance data, so we have to update $data ourselves
-function fixPrepatch(to, ___) {
```

Comment:
We would early exit because `_dataRefresh` was never `true`. Because we'd always exit early from the `// Call fetch hooks on components matched by the route.` section.

This was changing behavior from standard vue and I verified that data was still loading by going to two child routes which shared the same components. More specifically, two list pages which hadn't overridden the default (a few random ones under more resources).